### PR TITLE
Cleanup/reduce map

### PR DESCRIPTION
**Pull request by rosecers**
This is a PR focused on re-allocating code in src/map/map.ts to other modules where possible to reduce the size of src/map/map.ts (currently ~1200 lines). This is in response to #60 but also ongoing conversation.

First moves:
Making a class `src/marker/marker.ts` where all marker functions can live. Difficulty is that many marker functions rely on the map._plot aspect, so this only reduced `map.ts` by about 70 lines.
Moving _option-only dependent functions to src/map/options.ts. This was more fruitful (~150 lines), but may cause problems if there are functions which are *this* map specific and options is being used for multiple maps.

### CODE REVIEW

File: src/map/map.ts [hunk 1]
```diff
@@ -284,20 +249,26 @@ export class PropertiesMap {
...
+        data.activate();
 
         if (this._is3D()) {
             this._restyle({'marker.size': this._sizes(1)} as Data, 1);
         }
+
+        const activeData = this._selected.get(guid);
+        assert(activeData !== undefined);
+        const activeIndexes = this._indexer.from_environment(activeData.current);
+        this.activeChanged(guid, activeIndexes);
```
Comment (Luthaf) [inline src/map/map.ts:271]: why are you calling `activeChanged` here? 

The goal for this callback is to be called when the user change the active marker by interacting with the map; while the goal of `setActive` is to change the active marker on the map when the user interacted with another panel.

File: src/map/map.ts [hunk 2]
```diff
@@ -1138,84 +1008,53 @@ export class PropertiesMap {
...
-              marker.style.right = `${plotWidth - x}px`;
-          }
+        if (!this._is3D()) {this._updateMarkers(); }
     }
 
     /**
      * Update the position, color & size of all markers in the map
      */
-    private _updateAllMarkers(): void {
+    private _updateMarkers(data: MarkerData[] = Array.from(this._selected.values())): void {
```
Comment (Luthaf) [inline src/map/map.ts:1031]: The doc comment no longer correspond to the functions

File: src/map/map.ts [hunk 3]
```diff
@@ -1138,84 +1008,53 @@ export class PropertiesMap {
...
         if (this._is3D()) {
+            for (const datum of data) {
```
Comment (Luthaf) [inline src/map/map.ts:1033]: could also be `data.forEach((d) => d.toggleVisible(false))` if you want

File: src/map/marker.ts [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    public color: string;
+    /// index of the currently displayed structure
+    public current: number;
+    /// Marker indicating the position of the point in 2D mode
+    /// Using an HTML element is much faster than using Plolty to restyle the full plot,
+    /// especially with more than 100k points. In 3D mode, a separate trace is
+    /// used instead.
+    public marker: HTMLElement;
+
+    constructor(
```
Comment (Luthaf) [inline src/map/marker.ts:17]: All functions need documentation here

File: src/map/marker.ts [hunk 5]
```diff
@@ -0,0 +1,75 @@
+import assert from 'assert';
+
+import {EnvironmentIndexer, Indexes} from '../indexer';
+
+/// interface to contain synchronized parameters for marker instances
```
Comment (Luthaf) [inline src/map/marker.ts:5]:
```suggestion
/// Data associated with markers on the map indicating selected environments/structures
```

File: src/map/map.ts (hunk 3 above)
Comment (rosecers) [reply src/map/map.ts:1062]: Yes, yes I do.

**Comment by Luthaf** [comment]
Tests are failling due to formatter checks, you can reformat the code with `npm run pretty`. Please also install a https://prettier.io/ plugin in your editor to automatically format files as you edit them, this will reduce diff noise!

**Comment by Luthaf** [comment]
It's going live! 🚀

**Pull request merged**